crates/hir_expand/src/db.rs

```diff
@@ -13,6 +13,12 @@ use crate::{
     MacroFile, ProcMacroExpander,
 };
 
+/// Total limit on the number of tokens produced by any macro invocation.
+///
+/// If an invocation produces more tokens than this limit, it will not be stored in the database and
+/// an error will be emitted.
+const TOKEN_LIMIT: usize = 0x80000;
```

Comment:
Let's use decimal limit, like we do for other limits (and we probably should add a `limits.rs` file to each relevant crate to centralize them...).

But, more generally, macro limiting is hard. The bad case is this one:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=3c381dc730c2d0f73a644d1bd9c96994

```rust 
macro_rules! zip_bomb {
    ($($x:tt)*) => {
        zip_bomb!($($x)* $($x)*); 
    }
}

zip_bomb!(nope);
```

Kinda hard to contain without placing a linear limit on the size of the tree.